### PR TITLE
Update readme process in README.md

### PR DESCRIPTION
**Pull request by ericgraf**
Quick update to the README file updating the release process.

### CODE REVIEW

**Review by MarcelMue — APPROVED** [review]
LGTM but your branch is out of date.

**Comment by stone-z** [comment]
I think it's good to bring this up to date, and you can mention the steps here but I'd also say it's worth a link to the docs https://intranet.giantswarm.io/docs/dev-and-releng/how-to-release-a-project/

**Comment by MarcelMue** [comment]
> I think it's good to bring this up to date, and you can mention the steps here but I'd also say it's worth a link to the docs https://intranet.giantswarm.io/docs/dev-and-releng/how-to-release-a-project/

Not sure if we should link to private docs on a public project thouh :thinking:

**Comment by stone-z** [comment]
Right, not great for outside users I know, but non-GS can't release this project anyway. They can take the whole workflow and re-use it if they like, and a team was already looking at re-publishing the releasing a project docs on `docs` instead (so we can point to that once it happens). But just updating the instructions here has the same issue as before where we change the release process org-wide and then the in-repo docs are out of date.

**Comment by ericgraf** [comment]
Is our internal docs considered the source of truth and most up to date ? 

If it is, how about updating the readme to say
```
## Release Process

Currently only members of the Giantswarm org can create releases.
For up to date steps please referrer to internal docs here: https://intranet.giantswarm.io/docs/dev-and-releng/how-to-release-a-project/

```

**Comment by MarcelMue** [comment]
Fine with me to put that.

File: README.md [hunk 1]
```diff
@@ -48,17 +48,9 @@ We have this differentiation in place because in our Control Planes, we don't us
...
-* This will push a new git tag and trigger a new tarball to be pushed to the
-[default-catalog].  
-* Update [cluster-operator] with the new version.
-
-[app-operator]: https://github.com/giantswarm/app-operator
-[cluster-operator]: https://github.com/giantswarm/cluster-operator
-[default-catalog]: https://github.com/giantswarm/default-catalog
-[default-test-catalog]: https://github.com/giantswarm/default-test-catalog
+Currently only members of the Giantswarm org can create releases.
+For up to date steps please referrer to internal docs here: https://intranet.giantswarm.io/docs/dev-and-releng/how-to-release-a-project/
```
Comment (stone-z) [inline README.md:52]:
```suggestion
For up to date steps, please refer to internal docs here: https://intranet.giantswarm.io/docs/dev-and-releng/how-to-release-a-project/
```

File: README.md [hunk 2]
```diff
@@ -48,17 +48,9 @@ We have this differentiation in place because in our Control Planes, we don't us
...
-changelog entry.
-* This will push a new git tag and trigger a new tarball to be pushed to the
-[default-catalog].  
-* Update [cluster-operator] with the new version.
-
-[app-operator]: https://github.com/giantswarm/app-operator
-[cluster-operator]: https://github.com/giantswarm/cluster-operator
-[default-catalog]: https://github.com/giantswarm/default-catalog
-[default-test-catalog]: https://github.com/giantswarm/default-test-catalog
+Currently only members of the Giantswarm org can create releases.
```
Comment (stone-z) [inline README.md:51]:
```suggestion
Currently only members of the Giant Swarm organization can create releases for this project.
```

**Pull request merged**